core/templates/dev/head/services/alertsService.js

```diff
@@ -0,0 +1,166 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for handling warnings and info messages.
+ *
+ * @author sll@google.com (Sean Lip)
+ */
+
+oppia.factory('alertsService', ['$log', function($log) {
+  var alertsService = {
```

Comment:
I think it would be good to add some documentation here explaining the difference between the format of the different messages. If I understand correctly, warnings are simple strings, whereas messages are JS objects.

Alternatively, you could make the warnings have the same structure as the messages (and all of them have the same type). That might be conceptually easier.